Camel IT failing on Fedora + Bash due to environment variables containing bash scripts with variables

On Fedora + Bash, the Camel IT tests are failing with this error:
```
Caused by: java.util.NoSuchElementException: Property IFS+x not found
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:89)
	at io.quarkus.runtime.configuration.ConfigExpander.accept(ConfigExpander.java:55)
	at io.quarkus.runtime.configuration.ConfigExpander.accept(ConfigExpander.java:15)
	at org.wildfly.common.expression.ExpressionNode.emit(ExpressionNode.java:42)
	at org.wildfly.common.expression.CompositeNode.emit(CompositeNode.java:42)
	at org.wildfly.common.expression.Expression.evaluateException(Expression.java:75)
	at org.wildfly.common.expression.Expression.evaluate(Expression.java:89)
	at io.quarkus.runtime.configuration.ExpandingConfigSource.expand(ExpandingConfigSource.java:46)
	at io.quarkus.runtime.configuration.ExpandingConfigSource.getValue(ExpandingConfigSource.java:38)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:74)
	at io.quarkus.camel.core.deployment.CamelIntiProcessor.createInitiTask(CamelIntiProcessor.java:61)
```

This is due to the `BASH_FUNC_*` environment properties containing things like that:
```
if [ -n "${IFS+x}" ]; then
```
as MP Config tries to expand the variable and fail.

What makes the Camel ITs specific is that they iterate over all the configuration properties to push them to Camel.

--- a/extensions/camel/camel-core/deployment/src/main/java/io/quarkus/camel/core/deployment/CamelInitProcessor.java
+++ b/extensions/camel/camel-core/deployment/src/main/java/io/quarkus/camel/core/deployment/CamelInitProcessor.java
@@ -57,8 +57,13 @@ class CamelInitProcessor {
     CamelRuntimeBuildItem createInitTask(RecorderContext recorderContext, CamelTemplate template) {
         Properties properties = new Properties();
         Config config = ConfigProvider.getConfig();
-        for (String i : config.getPropertyNames()) {
-            properties.put(i, config.getValue(i, String.class));
+        for (String property : config.getPropertyNames()) {
+            if (property.startsWith("camel.")) {
+                properties.put(property, config.getValue(property, String.class));
+            }
+            if (property.startsWith("integration.")) {
+                properties.put(property.substring("integration.".length()), config.getValue(property, String.class));
+            }
         }
 
         String clazz = properties.getProperty(CamelRuntime.PROP_CAMEL_RUNTIME, CamelRuntime.class.getName());

--- a/extensions/camel/camel-core/runtime/src/main/java/io/quarkus/camel/core/runtime/CamelRuntime.java
+++ b/extensions/camel/camel-core/runtime/src/main/java/io/quarkus/camel/core/runtime/CamelRuntime.java
@@ -28,6 +28,7 @@ import org.apache.camel.RuntimeCamelException;
 import org.apache.camel.component.properties.PropertiesComponent;
 import org.apache.camel.impl.AbstractCamelContext;
 import org.apache.camel.impl.DefaultCamelContext;
+import org.apache.camel.model.ModelCamelContext;
 import org.apache.camel.model.ModelHelper;
 import org.apache.camel.model.RouteDefinition;
 import org.apache.camel.support.ResourceHelper;
@@ -37,20 +38,18 @@ import org.apache.camel.util.ObjectHelper;
 public class CamelRuntime extends ServiceSupport {
 
     public static final String PFX_CAMEL = "camel.";
-    public static final String PFX_CAMEL_PROPERTIES = PFX_CAMEL + "properties.";
+    public static final String PFX_CAMEL_PROPERTIES = PFX_CAMEL + "component.properties.";
     public static final String PFX_CAMEL_CONTEXT = PFX_CAMEL + "context.";
 
     public static final String PROP_CAMEL_RUNTIME = PFX_CAMEL + "runtime";
-    public static final String PROP_CAMEL_CONF = PFX_CAMEL + "conf";
-    public static final String PROP_CAMEL_CONFD = PFX_CAMEL + "confd";
-    public static final String PROP_CAMEL_DUMP = PFX_CAMEL + "dump";
-    public static final String PROP_CAMEL_DEFER = PFX_CAMEL + "defer";
+    public static final String PROP_CAMEL_ROUTES = PFX_CAMEL + "routes.";
+    public static final String PROP_CAMEL_ROUTES_DUMP = PROP_CAMEL_ROUTES + "dump";
+    public static final String PROP_CAMEL_ROUTES_LOCATIONS = PROP_CAMEL_ROUTES + "locations";
 
     protected RuntimeRegistry registry;
     protected Properties properties;
     protected AbstractCamelContext context;
     protected List<RoutesBuilder> builders;
-    protected PropertiesComponent propertiesComponent;
 
     public void bind(String name, Object object) {
         registry.bind(name, object);
@@ -62,10 +61,8 @@ public class CamelRuntime extends ServiceSupport {
 
     public void doInit() {
         try {
-            AbstractCamelContext context = createContext();
-            context.setRegistry(registry);
-            context.setAutoStartup(false);
-            this.context = context;
+            this.context = createContext();
+            this.context.setRegistry(registry);
 
             // Configure the camel context using properties in the form:
             //
@@ -73,16 +70,8 @@ public class CamelRuntime extends ServiceSupport {
             //
             RuntimeSupport.bindProperties(properties, context, PFX_CAMEL_CONTEXT);
 
+            context.setLoadTypeConverters(false);
             context.getModelJAXBContextFactory().newJAXBContext();
-
-            propertiesComponent = new PropertiesComponent();
-            propertiesComponent.setInitialProperties(properties);
-            RuntimeSupport.bindProperties(properties, propertiesComponent, PFX_CAMEL_PROPERTIES);
-            context.addComponent("properties", propertiesComponent);
-
-            loadRoutesFromBuilders(true);
-
-            context.start();
         } catch (Exception e) {
             throw RuntimeCamelException.wrapRuntimeCamelException(e);
         }
@@ -91,21 +80,17 @@ public class CamelRuntime extends ServiceSupport {
     public void doStart() throws Exception {
         log.info("Apache Camel {} (CamelContext: {}) is starting", context.getVersion(), context.getName());
 
-        String conf = getProperty(PROP_CAMEL_CONF);
-        String confd = getProperty(PROP_CAMEL_CONFD);
-        log.info("confPath: {}", conf);
-        log.info("confDPath: {}", confd);
-        RuntimeSupport.loadConfigSources(properties, conf, confd);
+        PropertiesComponent pc = createPropertiesComponent(properties);
+        RuntimeSupport.bindProperties(pc.getInitialProperties(), pc, PFX_CAMEL_PROPERTIES);
+        context.addComponent("properties", pc);
 
-        loadRoutesFromBuilders(false);
-        loadRoutes();
+        configureContext(context);
+        loadRoutes(context);
 
-        if (Boolean.parseBoolean(getProperty(PROP_CAMEL_DUMP))) {
+        context.start();
+
+        if (Boolean.parseBoolean(getProperty(PROP_CAMEL_ROUTES_DUMP))) {
             dumpRoutes();
-        }
-
-        for (Route route : getContext().getRoutes()) {
-            getContext().getRouteController().startRoute(route.getId());
         }
     }
 
@@ -114,32 +99,28 @@ public class CamelRuntime extends ServiceSupport {
         context.shutdown();
     }
 
-    protected void loadRoutesFromBuilders(boolean initPhase) throws Exception {
-        if (builders != null && !builders.isEmpty()) {
-            boolean defer = Boolean.parseBoolean(getProperty(PROP_CAMEL_DEFER));
-            if (defer ^ initPhase) {
-                for (RoutesBuilder b : builders) {
-                    if (b instanceof RouteBuilderExt) {
-                        ((RouteBuilderExt) b).setRegistry(registry);
-                    }
-                    context.addRoutes(b);
-                }
+    protected void loadRoutes(CamelContext context) throws Exception {
+        for (RoutesBuilder b : builders) {
+            if (b instanceof RouteBuilderExt) {
+                ((RouteBuilderExt) b).setRegistry(registry);
             }
+            context.addRoutes(b);
         }
-    }
 
-    protected void loadRoutes() throws Exception {
-        String routesUri = getProperty("camel.routesUri");
-        log.info("routesUri: {}", routesUri);
+        String routesUri = getProperty(PROP_CAMEL_ROUTES_LOCATIONS);
         if (ObjectHelper.isNotEmpty(routesUri)) {
+            log.info("routesUri: {}", routesUri);
+
+            ModelCamelContext mcc = context.adapt(ModelCamelContext.class);
+
             try (InputStream is = ResourceHelper.resolveMandatoryResourceAsInputStream(getContext(), routesUri)) {
-                context.addRouteDefinitions(context.loadRoutesDefinition(is).getRoutes());
+                mcc.addRouteDefinitions(mcc.loadRoutesDefinition(is).getRoutes());
             }
         }
     }
 
     protected String getProperty(String name) throws Exception {
-        return propertiesComponent.parseUri(propertiesComponent.getPrefixToken() + name + propertiesComponent.getSuffixToken());
+        return context.resolvePropertyPlaceholders(context.getPropertyPrefixToken() + name + context.getPropertySuffixToken());
     }
 
     protected DefaultCamelContext createContext() {
@@ -160,6 +141,18 @@ public class CamelRuntime extends ServiceSupport {
 
     public CamelContext getContext() {
         return context;
+    }
+
+    protected PropertiesComponent createPropertiesComponent(Properties initialPoperties) {
+        PropertiesComponent pc = new PropertiesComponent();
+        pc.setInitialProperties(initialPoperties);
+
+        RuntimeSupport.bindProperties(properties, pc, PFX_CAMEL_PROPERTIES);
+
+        return pc;
+    }
+
+    protected void configureContext(CamelContext context) {
     }
 
     protected void dumpRoutes() {

--- a/extensions/camel/camel-core/runtime/src/main/java/io/quarkus/camel/core/runtime/RuntimeSupport.java
+++ b/extensions/camel/camel-core/runtime/src/main/java/io/quarkus/camel/core/runtime/RuntimeSupport.java
@@ -16,18 +16,10 @@
  */
 package io.quarkus.camel.core.runtime;
 
-import java.io.IOException;
-import java.io.Reader;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.List;
 import java.util.Properties;
-import java.util.stream.Collectors;
 
 import org.apache.camel.CamelContext;
 import org.apache.camel.support.IntrospectionSupport;
-import org.apache.camel.util.ObjectHelper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -35,36 +27,6 @@ public final class RuntimeSupport {
     private static final Logger LOGGER = LoggerFactory.getLogger(RuntimeSupport.class);
 
     private RuntimeSupport() {
-    }
-
-    public static void loadConfigSources(Properties properties, String conf, String confd) {
-        // Main location
-        if (ObjectHelper.isNotEmpty(conf)) {
-            try (Reader reader = Files.newBufferedReader(Paths.get(conf))) {
-                properties.load(reader);
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        }
-
-        // Additional locations
-        if (ObjectHelper.isNotEmpty(confd)) {
-            Path root = Paths.get(confd);
-            if (Files.exists(root)) {
-                try {
-                    List<Path> paths = Files.walk(root)
-                            .filter(p -> p.getFileSystem().toString().endsWith(".properties"))
-                            .collect(Collectors.toList());
-                    for (Path path : paths) {
-                        try (Reader reader = Files.newBufferedReader(path)) {
-                            properties.load(reader);
-                        }
-                    }
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
-            }
-        }
     }
 
     public static void bindProperties(CamelContext context, Properties properties, Object target, String prefix) {

--- a/integration-tests/camel-core/src/main/java/io/quarkus/camel/it/core/CamelServlet.java
+++ b/integration-tests/camel-core/src/main/java/io/quarkus/camel/it/core/CamelServlet.java
@@ -12,16 +12,12 @@ import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 
 import org.apache.camel.Route;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import io.quarkus.camel.core.runtime.CamelRuntime;
 
 @Path("/")
 @ApplicationScoped
 public class CamelServlet {
-    private static final Logger LOGGER = LoggerFactory.getLogger(CamelServlet.class);
-
     @Inject
     CamelRuntime runtime;
 

--- a/integration-tests/camel-core/src/main/resources/application.properties
+++ b/integration-tests/camel-core/src/main/resources/application.properties
@@ -1,9 +1,22 @@
+#
+# Quarkus
+#
+quarkus.log.file.enable = false
+
+#
+# Camel
+#
 camel.context.name=quarkus-camel-example
-camel.properties.prefixToken={{
-camel.properties.suffixToken=}}
-camel.conf={{env:CAMEL_CONF:}}
-camel.confd={{env:CAMEL_CONFD:}}
-camel.dump=false
-camel.defer=false
-camel.routesUri=
-folder={{sys:folder:./target/orders}}
+camel.context.loadTypeConverters=false
+
+camel.routes.dump=true
+camel.routes.defer=true
+camel.routes.locations=
+
+camel.component.properties.prefixToken={{
+camel.component.properties.suffixToken=}}
+
+#
+# Integration
+#
+integration.folder={{sys:folder:./target/orders}}

--- a/integration-tests/camel-salesforce/pom.xml
+++ b/integration-tests/camel-salesforce/pom.xml
@@ -127,7 +127,6 @@
             </plugin>
         </plugins>
     </build>
-<!--
     <profiles>
         <profile>
             <id>native-image</id>
@@ -181,6 +180,5 @@
             </build>
         </profile>
     </profiles>
--->
 
 </project>

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -51,10 +51,7 @@
         <module>spring-di</module>
         <module>infinispan-cache-jpa</module>
         <module>infinispan-cache-jpa-stress</module>
-        <!-- Camel tests are not working on Fedora + Bash
-        see https://github.com/quarkusio/quarkus/issues/1130
         <module>camel-core</module>
         <module>camel-salesforce</module>
-        -->
     </modules>
 </project>

